pkg/appgateserver/docs/server.md

```diff
@@ -0,0 +1,278 @@
+
+In order to run an `AppGateServer`, the user must first stake either their
+application or gateway accounts.
+
+_NOTE: The `AppGateServer` will still start without the actor being staked but
+will not be usable_
+
+### Applications
+
+To start the `AppGateServer` as an application the following command must be ran:
+
+```sh
+poktrolld --home={CHAIN_HOME_DIR} appgate-server --signing-key={SIGNING KEY NAME} \
+    --self-signing --listening-endpoint={ENDPOINT TO LISTEN ON} \
+    --query-node={ENDPOINT OF A POCKET NODE TO QUERY ON CHAIN DATA} \
+    --keyring-backend={KEYRING}
+```
+
+Here the `signing-key` is the name of the key in the `keyring-backend` keyring.
+This is used to sign relays. If the `self-signing` flag is set this means the
+`AppGateServer` will store the address associated with the `signing-key`'s key
+and use this to retrieve its own ring and sign relays using its own key and ring.
+
+The `query-node` flag is the flag of a trusted Pocket node that will be used to
+query the on-chain state for session retrieval, ring creation and public key
+retrieval for response verification.
+
+The `listening-endpoint` flag determines the endpoint the `AppGateServer` will
+listen on to receive requests from the application. This is the endpoint the
+application will send their requests to.
+
+### Gateways
+
+Starting an `AppGateServer` as a gateway works similarly to starting an
+[application](#applications). The only difference is the omission of the
+`self-signing` flag.
+
+```sh
+poktrolld --home={CHAIN_HOME_DIR} appgate-server --signing-key={SIGNING KEY NAME} \
+    --listening-endpoint={ENDPOINT TO LISTEN ON} \
+    --query-node={ENDPOINT OF A POCKET NODE TO QUERY ON CHAIN DATA} \
+    --keyring-backend={KEYRING}
+```
+
+When started without the `self-signing` flag the `AppGateServer` will not store
+the address associated with the `signing-key` flag. This means the requests that
+it receives **must** include the application's address in the
+[endpoint](#requests).
+
+When a gateway receives a request it will fetch the ring of the application
+sending the request. It will then sign the request using `signing-key` used to
+start the gateway.
+
+_NOTE: The gateway is only able to sign relays for applications who have
+delegated to the gateway_
+
+## Requests
+
+Requests from an application are received on the `AppGateServer`'s listening
+address. This acts as a single endpoint the application can send their requests
+to.
+
+When sending a request the service **must** be specified in the endpoint and
+if the `AppGateServer` is a gateway then the application's address must also be
+included.
+
+The endpoint will look like the following:
+
+```
+scheme://host:port/{service}[?senderAddr={application address}]
+```
+
+### Request Types
+
+Depending on the service the request will be of a different form: JSON-RPC,
+REST, gRPC, GraphQL, etc. These different payload types can be catagorised into
+two catagories.
+
+1. Symmetric Protocols
```

Comment:
@red-0ne @h5law **IMPORTANT COMMENT.**

I think this terminology is important so I wanted to understand if this is something we coined or if it's a standard?

---

[1] https://en.wikipedia.org/wiki/RPyC

I could insinuate was symmetric/assymetric protocols are but did a bit of research to see if it conflicts with industry standards. [1]
![Screenshot 2023-11-20 at 8 01 20 PM](https://github.com/pokt-network/poktroll/assets/1892194/30904d62-d636-46a7-926f-8da40cfd595a)
![Screenshot 2023-11-20 at 7 59 20 PM](https://github.com/pokt-network/poktroll/assets/1892194/7ab522b9-590b-4a00-99a2-7cb978172501)

I think we should call it `Synchronous` and `Asynchronous`

![Screenshot 2023-11-20 at 8 03 48 PM](https://github.com/pokt-network/poktroll/assets/1892194/522fb6aa-a30f-43d4-9772-c17e2be41d94)